### DISCUSSION TITLE
Replacing a CC2652P with an identical stick requires repair ?

### DISCUSSION THREAD

**Discussion by realroywalker** [Koenkk/zigbee2mqtt#7143] [Devices]
Hey all,

I have a CC2652P (Ebyte E72-2G4M20S1E) connected with an ESP32 (Socat), set up with the Zigbee2MQTT firmware, it works OK but occasional WiFi drops cause problems.
Because of this I wanted to swap it out with a new stick I made that is USB instead, but uses the same Ebyte E72-2G4M20S1E board it is using this design: https://github.com/egony/cc2652p_E72-2G4M20S1E so I used the Egony firmware (which I believe is just patched Zigbee2MQTT firmware so that LEDs work).

When I stop my Z2M instance, and swap the config to use the USB stick instead, it starts fine, but I cannot control any of my devices (timeout when turning on a light for example).
Can anyone explain if repairing all devices is required for this scenario ? - From what I read here (https://www.zigbee2mqtt.io/information/FAQ.html#what-does-and-does-not-require-repairing-of-all-devices) it seemed to suggest not?

**Comment by gio-dot** [comment]
You have to erase Nvram of the Usb stick before. In this way It can load previous coordinator backup and start without the need to repair.
You have to delete all nvram with the next script: zStackEraseAllNvMem.js
It is already in your zigbee2mqtt/scripts folder. Simply run it

**Reply by Koenkk** [reply]
Besides this, reflashing will also do the trick.

**Reply by realroywalker** [reply]
Thanks, this is where I'm a little confused as I did read that tip elsewhere (think it was on the FAQ). 
My stick was new (made myself) and fresh flashed with the firmware, from this I assumed the nvram doesn't need erasing since it's never been used before?

**Reply by Koenkk** [reply]
It depends wether the stick has already been started once. This will not be required anymore after https://github.com/Koenkk/zigbee-herdsman/pull/303 is merged.

**Reply by gio-dot** [reply]
This change also the migration from cc2531 to 2652? Old cc2531 Coordinator backup can be used by 2652? I'm understanding well?

**Reply by Koenkk** [reply]
Yes

**Reply by realroywalker** [reply]
Interesting link, thanks for the info. 
Odd thing this was indeed the very first time the chip was used straight to my server with my existing Z2M instance.

I will try doing the nvram reset though once the wife and kids go out and see if it restores ok 😁
